src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentCompletionEndpointTest.cs

```diff
@@ -0,0 +1,508 @@
+
+        VSInternalCompletionList? response = null;
+        if (delegatedItemLabels is not null)
+        {
+            response = new VSInternalCompletionList()
+            {
+                Items = delegatedItemLabels.Select((label) => new VSInternalCompletionItem()
+                {
+                    Label = label,
+                    CommitCharacters = delegatedItemCommitCharacters,
+                    // If test specifies not to commit with space, set kind to element since we remove space
+                    // commit from elements only. Otherwise test doesn't care, so set to None
+                    Kind = !commitElementsWithSpace ? CompletionItemKind.Element : CompletionItemKind.None,
+                }).ToArray(),
+                IsIncomplete = true
+            };
+        }
+
+        var requestInvoker = new TestLSPRequestInvoker([(Methods.TextDocumentCompletionName, response)]);
+
+        var snippetCompletionItemProvider = new SnippetCompletionItemProvider(new SnippetCache());
+        if (snippetLabels is not null)
+        {
+            var snippetInfos = snippetLabels.Select(label => new SnippetInfo(label, label, label, string.Empty, SnippetLanguage.Html)).ToImmutableArray();
+            snippetCompletionItemProvider.SnippetCache.Update(SnippetLanguage.Html, snippetInfos);
+        }
+
+        var completionSetting = new CompletionSetting
+        {
+            CompletionItem = new CompletionItemSetting(),
+            CompletionItemKind = new CompletionItemKindSetting()
+            {
+                ValueSet = (CompletionItemKind[])Enum.GetValues(typeof(CompletionItemKind)),
+            },
+            CompletionListSetting = new CompletionListSetting()
+            {
+                ItemDefaults = ["commitCharacters", "editRange", "insertTextFormat"]
+            },
+            ContextSupport = false,
+            InsertTextMode = InsertTextMode.AsIs,
+        };
+        var clientCapabilities = new VSInternalClientCapabilities
+        {
+            TextDocument = new TextDocumentClientCapabilities
+            {
+                Completion = completionSetting
+            }
+        };
+        var endpoint = new CohostDocumentCompletionEndpoint(
+            RemoteServiceInvoker,
+            clientSettingsManager,
+            TestHtmlDocumentSynchronizer.Instance,
+            snippetCompletionItemProvider,
+            requestInvoker,
+            LoggerFactory);
+        endpoint.GetTestAccessor().SetClientCapabilities(clientCapabilities);
+
+        var request = new RoslynCompletionParams()
+        {
+            TextDocument = new RoslynTextDocumentIdentifier()
+            {
+                Uri = document.CreateUri()
+            },
+            Position = RoslynLspExtensions.GetPosition(sourceText, input.Position),
+            Context = completionContext
+        };
+
+        // Roslyn doesn't always return all items right away, so using retry logic
```

Comment:
This is very surprising. Is this copied from the existing tests, or is this something you are seeing in these tests specifically?

Reply:
I did see it in these tests specifically. E.g. at the "@" I initially got something like 556 items, but later 1000 items (max items returned by Roslyn). Adding re-try made tests run reliably. I am not sure why.

Reply:
Seems like that could be an issue with our test code, but not sure. Were the items we were expecting not present when it return 556? Seems like a fair number of completion items

Reply:
Removed item count verification and retries per our offline online conversation. :)